infrastructure/permission/validPermissions.go

```diff
@@ -0,0 +1,94 @@
+package permission
+
+import "github.com/ystv/web-auth/permission/permissions"
+
+func GetValidPermissions(p permissions.Permissions) (m map[string]bool) {
```

Comment:
I'm still not sure what this function does -  as far as I can tell, it returns the permissions that a user can also have to do an action that requires `p` (eg to do "Calendar.Meeting.Creator" actions they can also have "Calendar.Admin"), is that right? If so, it may be worth trying to come up with another name, because `GetValidPermissions` makes me think that it returns _all_ the valid permission names, or somehow checks that a permission name is valid.

Also, what's the significance of the return type being `map[string]bool` rather than `map[Permissions]bool`?

Reply:
The permission gets the lowest permission to do a task and then checks if you have the required permissions, this doesn't give all permissions but checks if your permissions matches the valid permissions to do something, that is why `SuperUser` is out of the switch as `SuperUser` can do any of the roles, except for `MenuDisable`

I think it was used with strings rather than permission class so was just more convenient, plus if a description is changed then that would change this I think

Reply:
I see - I still think it might be worth renaming this function, or at least adding a documentation comment, because it took me a while to understand what it does. Perhaps something like `SufficientPermissionsFor()`? Just a starting point, don't feel the need to go with that.

And I'm not entirely sure I understand the second part of your comment - surely if the string value changes, that's now a different permission (unless I've misunderstood what you mean by "description")?

Reply:
By description, I mean the description field of the permission, I have renamed the function and added a comment